app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt

```diff
@@ -1156,6 +1179,30 @@ class AuthRepositoryImpl(
     ): Boolean = passwordPolicies
         .all { validatePasswordAgainstPolicy(password, it) }
 
+    override suspend fun sendVerificationEmail(
+        email: String,
+        name: String,
+        receiveMarketingEmails: Boolean,
+    ): SendVerificationEmailResult =
+        identityService.sendVerificationEmail(
+            SendVerificationEmailRequestJson(
+                email = email,
+                name = name,
+                receiveMarketingEmails = receiveMarketingEmails,
+            )
+        ).fold(
+            onSuccess = {
+                SendVerificationEmailResult.Success(it?.string())
+            },
+            onFailure = {
+                // error throw in [ResultCall] if response body is null
+                if (it is IllegalStateException) {
+                    return SendVerificationEmailResult.Success(null)
+                }
```

Comment:
We should not need scalars at all Kotlin.SerializationX should support this.

I am curious if you actually saw this exception, since the `ResponseBody` should never be null even if the body itself is null.

Reply:
yes I saw the exception and had to dig into the code to find out what was going on. ResultCall is throwing this exception if body is null, which it is, in this case is a 204 returned from the server.

Reply:
I see, I think that is a ramification of an API that can sometimes return a 200 with payload and a 204 without payload.

I have not seen this type of pattern before. Do we know why this API functions this way?

Reply:
Not sure on why it was done this way, but the 200 has a token response to handle and we can navigate to `CompleteRegistration` and the 204 we need to navigate to `CheckEmail`. But with you new changes to handle to null body I can discard this.